database/cassandra/cassandra.go

```diff
@@ -197,12 +197,9 @@ func (c *Cassandra) Run(migration io.Reader) error {
 }
 
 func (c *Cassandra) SetVersion(version int, dirty bool) error {
-	query := `TRUNCATE "` + c.config.MigrationsTable + `"`
-	if err := c.session.Query(query).Exec(); err != nil {
-		return &database.Error{OrigErr: err, Query: []byte(query)}
-	}
+
 	if version >= 0 {
-		query = `INSERT INTO "` + c.config.MigrationsTable + `" (version, dirty) VALUES (?, ?)`
+		query := `INSERT INTO "` + c.config.MigrationsTable + `" (version, dirty) VALUES (?, ?)`
```

Comment:
This needs to be an upsert. Azure Cosmos DB may not support `TRUNCATE`, but Cassandra does so any change you make needs to be compatible with Cassandra. Another option would be to fork this driver and create an Azure Cosmos DB specific one.